ui/dashboards/src/components/GridLayout/GridContainer.tsx

```diff
@@ -0,0 +1,221 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useEffect, useState } from 'react';
+import { styled } from '@mui/material';
+
+export interface GridContainerProps {
+  children: React.ReactNode;
+}
+
+export function GridContainer(props: GridContainerProps) {
+  const [isFirstRender, setIsFirstRender] = useState(true);
```

Comment:
what about using `useRef` here so that GridContainer doesn't render again when we set it back to false?

Reply:
or use [this](https://usehooks-ts.com/react-hook/use-is-first-render)?

Reply:
Yeah, the problem is, we _do_ want it to render twice (which a ref will prevent). Otherwise animation/CSS transform stays disabled until something else triggers it to render again.

Reply:
Hahaha, I thought we wanted to disable animation when the grid first mounts? Right now because of some of that weird state stuff (which maybe you fixed in #711 ?) changing template variables recreates the store and so the grids get remounted, causing them to do that "grow" animation from the top-left.

The reason we don't want the CSS transforms disabled completely (which would kill all the animation I think), is because of this from the docs:

![Screen Shot 2022-10-31 at 9 38 41 AM](https://user-images.githubusercontent.com/428023/199048164-1c423004-f54a-44a0-91e5-cfe969929f4b.png)

When someone is dragging/resizing (after the initial render), I think we probably want that perf boost.

Reply:
Right, my PR fixes the "grow" animation issue. Sorry i'm getting confused haha. Why do we want to disable animation on first mount though?

Reply:
All right, if your PR fixed it, then maybe this isn't necessary? I'm gonna go try out your PR. I thought we didn't want that animation transition at all when grid items first render.